### PR TITLE
initial cut at a standalone TV tool for bitcode files

### PR DESCRIPTION
**Pull request by regehr**

### CODE REVIEW

File: tools/alive-tv.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  std::unique_ptr<llvm::Module> M =
+    ExitOnErr(getOwningLazyBitcodeModule(std::move(MB), Context,
+                                         /*ShouldLazyLoadMetadata=*/true));
+  ExitOnErr(M->materializeAll());
+  return M;
+}
+
+static llvm::Function *getSingleFunction(llvm::Module *M) {
+  llvm::Function *Ret = 0;
+  for (auto &F : *M) {
```
Comment (LebedevRI) [inline tools/alive-tv.cpp]: This only looks at the actual function definition, it ignores the declarations?
I.e. this should be just one function:
https://godbolt.org/z/RhmJqc

File: tools/alive-tv.cpp [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+#include <string_view>
+#include <vector>
+
+using namespace IR;
+using namespace tools;
+using namespace util;
+using namespace std;
+using namespace llvm_util;
+
+static llvm::cl::opt<std::string>
```
Comment (LebedevRI) [inline tools/alive-tv.cpp]: Would be useful to add these to a new standalone category
```
static llvm::cl::OptionCategory AliveTV("alive-tv options");

static llvm::cl::opt<std::string>
opt_file1(llvm::cl::Positional, llvm::cl::desc("first_bitcode_file"),
    llvm::cl::Required, llvm::cl::value_desc("filename"),
    cl::cat(AliveTV));
```

File: tools/alive-tv.cpp [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+static llvm::ExitOnError ExitOnErr;
+
+// adapted from llvm-dis.cpp
+static std::unique_ptr<llvm::Module> openInputFile(llvm::LLVMContext &Context,
+                                                   std::string InputFilename) {
+  std::unique_ptr<llvm::MemoryBuffer> MB =
+    ExitOnErr(errorOrToExpected(llvm::MemoryBuffer::getFile(InputFilename)));
+  std::unique_ptr<llvm::Module> M =
+    ExitOnErr(getOwningLazyBitcodeModule(std::move(MB), Context,
```
Comment (LebedevRI) [inline tools/alive-tv.cpp]: Hmm, is it too much to ask for it to just natively handle the `.ll`, too (or it does already) ? :)
I kinda expect to be heavily used, much like the `alive` with sudo-ir

File: tools/alive-tv.cpp (hunk 1 above)
Comment (regehr) [reply tools/alive-tv.cpp]: sorry, I don't understand what you're getting at.
I only want to look at the function definition, and I only want to look at modules containing a single such definition.

File: tools/alive-tv.cpp (hunk 2 above)
Comment (regehr) [reply tools/alive-tv.cpp]: yeah, I'll do this

File: tools/alive-tv.cpp (hunk 3 above)
Comment (regehr) [reply tools/alive-tv.cpp]: do you have a pointer to a utility function that handles .bc and .ll? this must exist somewhere...

File: tools/alive-tv.cpp (hunk 1 above)
Comment (LebedevRI) [reply tools/alive-tv.cpp]: That was the question :)
I was just checking that understood correctly that it won't error-out on https://godbolt.org/z/RhmJqc

Comment (regehr) [reply tools/alive-tv.cpp]: it errors out :)

Comment (LebedevRI) [reply tools/alive-tv.cpp]: Sounds like a bug then :)

Comment (regehr) [reply tools/alive-tv.cpp]: fixed thanks

File: tools/alive-tv.cpp (hunk 2 above)
Comment (regehr) [reply tools/alive-tv.cpp]: done

File: tools/alive-tv.cpp (hunk 3 above)
Comment (regehr) [reply tools/alive-tv.cpp]: let's leave this for now, it's not pressing and easy to add later

**Pull request merged**